resources/test-invoker.mjs

```diff
@@ -48,8 +67,51 @@ export class RAFTestInvoker extends TestInvoker {
     }
 }
 
+class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
+    _scheduleCallbacks(resolve) {
+        requestAnimationFrame(async () => {
+            await this._syncCallback();
+
+            let gotTimer = false;
+            let gotMessage = false;
+
+            const tryTriggerAsyncCallback = () => {
+                if (!gotTimer || !gotMessage)
+                    return;
+
+                this._asyncCallback();
+                setTimeout(async () => {
+                    await this._reportCallback();
+                    resolve();
+                }, 0);
+            };
+
+            setTimeout(() => {
+                gotTimer = true;
+                tryTriggerAsyncCallback();
+            });
+
+            const mc = new MessageChannel();
```

Comment:
It might be nice to use the same channel for all the tests, just to avoid extra garbage created by the runner. Not sure where there channel could live. Perhaps a static property on the class?

Reply:
are you thinking maybe something like:
```
class AsyncRAFTestInvoker extends BaseRAFTestInvoker {
    static mc = new MessageChannel();
    _scheduleCallbacks(resolve) {
        requestAnimationFrame(async () => {
            await this._syncCallback();

            let gotTimer = false;
            let gotMessage = false;

            const tryTriggerAsyncCallback = () => {
                if (!gotTimer || !gotMessage)
                    return;

                this._asyncCallback();
                setTimeout(async () => {
                    await this._reportCallback();
                    resolve();
                }, 0);
            };

            const handleMessage = () => {
                AsyncRAFTestInvoker.mc.port1.removeEventListener("message", handleMessage);
                gotMessage = true;
                tryTriggerAsyncCallback();
            };

            setTimeout(() => {
                gotTimer = true;
                tryTriggerAsyncCallback();
            });

            AsyncRAFTestInvoker.mc.port1.addEventListener("message", handleMessage);
            AsyncRAFTestInvoker.mc.port1.start();
            AsyncRAFTestInvoker.mc.port2.postMessage("speedometer");
        });
    }
}
```

Reply:
@smaug---- and following up here as well in regards to a static MessageChannel